src/main/scala/faces/utils/CorrespondenceMoMoRenderer.scala

```diff
@@ -0,0 +1,188 @@
+package faces.utils
+
+/*
+ * Copyright University of Basel, Graphics and Vision Research Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+import breeze.linalg.DenseVector
+import scalismo.common.PointId
+import scalismo.faces.color.{RGB, RGBA}
+import scalismo.faces.image.{PixelImage, PixelImageNormalization}
+import scalismo.faces.landmarks.TLMSLandmark2D
+import scalismo.faces.mesh.VertexColorMesh3D
+import scalismo.faces.momo.MoMo
+import scalismo.faces.parameters.{ParametricRenderer, RenderParameter}
+import scalismo.faces.render.{PixelShader, TriangleRenderer}
+import scalismo.faces.render.TriangleRenderer.TriangleFragment
+import scalismo.faces.sampling.face._
+import scalismo.geometry.{Point, Vector, _3D}
+import scalismo.mesh.{MeshSurfaceProperty, SurfacePointProperty}
+import scalismo.utils.Memoize
+
+import scala.collection.immutable
+
+/** parametric renderer for a Morphable Model, implements all useful Parameteric*Renderer interfaces */
+class CorrespondenceMoMoRenderer(val model: MoMo, val clearColor: RGBA)
+  extends ParametricImageRenderer[RGBA]
```

Comment:
Could we inherit from MoMoRenderer instead? Some functions are there already implemented as e.g. padCoefficients?

Reply:
Yes, I just committed the change.